tests/integration/conftest.py

```diff
@@ -0,0 +1,107 @@
+import warnings
+import pytest
+import requests
+import contextlib
+import time
+import functools
+from girder_worker.utils import JobStatus
+from datetime import timedelta, datetime
+
+
+@pytest.fixture(scope='module')
+def api_url():
+    def _api_url(uri=''):
+        return 'http://127.0.0.1:8989/api/v1/{}'.format(uri)
+    return _api_url
+
+
+# TODO: test with admin and non-admin user - are there (should there be)
+# differences between girder-worker functionality between the two?
+@pytest.fixture(scope='module')
+def session(api_url, request):
+    login = ('admin', 'letmein')
+    with requests.Session() as s:
+        try:
+            r = requests.get(api_url('user/authentication'),
+                             auth=login)
+        except requests.ConnectionError:
+            raise Exception(
+                'Unable to connect to {}.'.format(api_url()))
+
+        try:
+            s.headers['Girder-Token'] = r.json()['authToken']['token']
+        except KeyError:
+            raise Exception(
+                'Unable to login with user "{}", password "{}"'.format(*login))
+
+        yield s
+
+
+@pytest.fixture
+def wait_for(session, api_url):
+
+    @contextlib.contextmanager
+    def _wait_for(job_id, predicate, timeout=20, interval=0.3, on_timeout=None):
+        then = datetime.utcnow() + timedelta(seconds=timeout)
+        timeout = True
+
+        while datetime.utcnow() < then:
+            r = session.get(api_url('job/{}'.format(job_id)))
```

Comment:
I think the girder standard is to use % operator rather than format,  as format doesn't do coercion to Unicode automatically.